src/index.ts

```diff
@@ -1,65 +1,127 @@
+		// Unlike `ws.accept()`, `state.acceptWebSocket(ws)` informs the Workers Runtime that the WebSocket
+		// is "hibernatable", so the runtime does not need to pin this Durable Object to memory while
+		// the connection is open. During periods of inactivity, the Durable Object can be evicted
+		// from memory, but the WebSocket connection will remain open. If at some later point the
+		// WebSocket receives a message, the runtime will recreate the Durable Object
+		// (run the `constructor`) and deliver the message to the appropriate handler.
+		this.ctx.acceptWebSocket(server);
+
+		// Generate a random UUID for the session.
+		const id = crypto.randomUUID();
+
+		// Attach the session ID to the WebSocket connection and serialize it.
+		// This is necessary to restore the state of the connection when the Durable Object wakes up.
+		server.serializeAttachment({
+			...server.deserializeAttachment(),
+			id,
+		});
+
+		// Add the WebSocket connection to the map of active sessions.
+		this.sessions.set(server, { id });
+
+		return new Response(null, {
+			status: 101,
+			webSocket: client,
+		});
 	}
-}
 
-export default {
-	/**
-	 * This is the standard fetch handler for a Cloudflare Worker
-	 *
-	 * @param request - The request submitted to the Worker from the client
-	 * @param env - The interface to reference bindings declared in wrangler.toml
-	 * @param ctx - The execution context of the Worker
-	 * @returns The response to be sent back to the client
-	 */
-	async fetch(request, env, ctx): Promise<Response> {
-		// We will create a `DurableObjectId` using the pathname from the Worker request
-		// This id refers to a unique instance of our 'MyDurableObject' class above
-		let id: DurableObjectId = env.WEBSOCKET_HIBERNATION_SERVER.idFromName(new URL(request.url).pathname);
-
-		// This stub creates a communication channel with the Durable Object instance
-		// The Durable Object constructor will be invoked upon the first call for a given id
-		let stub = env.WEBSOCKET_HIBERNATION_SERVER.get(id);
-
-		// We call the `sayHello()` RPC method on the stub to invoke the method on the remote
-		// Durable Object instance
-		let greeting = await stub.sayHello('world');
-
-		return new Response(greeting);
-	},
-} satisfies ExportedHandler<Env>;
+	async webSocketMessage(sender: WebSocket, message: ArrayBuffer | string) {
+		// Get the session associated with the WebSocket connection.
+		const session = this.sessions.get(sender);
+
+		// Upon receiving a message from the client, the server replies with the same message, the session ID of the sender,
+		// and the total number of connections with the "[Durable Object]: " prefix
+		sender.send(`[Durable Object] message: ${message}, from: ${session.id}. Total connections: ${this.ctx.getWebSockets().length}`);
+
+		// Send a message to all WebSocket connections, loop over all the connected WebSockets.
+		// This wakes up all the connections and the DO is no longer in hibernation.
```

Comment:
`This wakes up all the connections and the DO is no longer in hibernation.`.

Well... not exactly.

1. If the DO was hibernating prior to receiving this WebSocket message, it is no longer hibernating now because we would've called the `constructor()` _first_, then called `webSocketMessage()`.
2. Inside of you `constructor()`, you call `this.ctx.getWebSockets()` to rebuild the `sessions` map. In doing this, you already recreate all the JS WebSockets.
3. Technically, you don't even need to use `this.ctx.getWebSockets()` here, because you could just use your `sessions` map. That might actually be even better, because then you could also include each connection's session ID in the `ws.send()`.